monai/handlers/mlflow_handler.py

```diff
@@ -196,4 +260,18 @@ def _default_iteration_log(self, engine: Engine) -> None:
         if not isinstance(loss, dict):
             loss = {self.tag_name: loss.item() if isinstance(loss, torch.Tensor) else loss}
 
-        mlflow.log_metrics(loss, step=engine.state.iteration)
+        if isinstance(engine, Trainer):
```

Comment:
Would that be a problem if the trainer is not from `monai.engine.Trainer`? For example, cell 7 in the [segmentation ignite notebook](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/unet_segmentation_3d_ignite.ipynb)

```python
trainer = ignite.engine.create_supervised_trainer(
    net, opt, loss, device, False
)
```

Reply:
Hi @mingxin-zheng ,
     Currently there will be a problem if the input engine is not a Trainer in monai engine. I will add some code here to ensure the engine type to be Trainer.
Thanks,
Bin